### ISSUE TITLE
XplatGenerateReleaseNotes support for TFS

### ISSUE THREAD

**Issue by gregpakes** [rfennell/vNextBuild#113]
### VSTS Extension you are using
XplatGenerateReleaseNotes

### Where are you running it?
TFS 2017 Update 1

### Version of Extension/Task
Latest

### Expected behavior and actual behavior
Extension uses API version **3.2-preview.1** which does not seem compatible with the latest TFS version.

When I run this locally, I get a 404.

**Comment by rfennell** [comment]
I need to check that, might be able to use an older version

**Comment by gregpakes** [comment]
FYI - the latest API version I have that works for the following endpoint is **3.1-preview.5**

> _apis/Release/releases/{ReleaseID}

**Comment by rfennell** [comment]
I have knocked back the API versions to ones that are on 2017.1 (and I think 2017RTM), as I am not doing anything that clever with API I did not think this would be an issue. 

So I have test branch that is still OK for VSTS but throwing an error on TFS, I needed to do a bit more debugging, I will keep the thread posted as to progress

**Comment by rfennell** [comment]
Have tested this on TFS 2017.1 and seems OK so have released it as 1.1.5

**Comment by gregpakes** [comment]
This still doesn't work for me.

```
2017-05-08T15:48:29.0811163Z Getting the details of release 684
2017-05-08T15:48:29.2839167Z ##[debug]task result: Failed
2017-05-08T15:48:29.2839167Z ##[debug]Processing: ##vso[task.issue type=error;]Unhandled: Unexpected token <
2017-05-08T15:48:29.2839167Z ##[error]Unhandled: Unexpected token <
2017-05-08T15:48:29.2839167Z ##[debug]Processing: ##vso[task.complete result=Failed;]Unhandled: Unexpected token <
2017-05-08T15:48:29.2995167Z ##[section]Finishing: Generate Release Notes based on Release Comparision API
```

**Comment by rfennell** [comment]
That is a different issue I think, as it has now managed to call the REST API to get the release, to we are calling a valid API version. but it looks like the results are not what we are expecting.

The call it is making is

```
https://myserver:8080/tfs/teamprojectname/_apis/Release/releases/684
```
Against api-version 3.1-preview.1

What do you see if you try this call in a browser?

**Comment by gregpakes** [comment]
It seems like the url is missing the collection name?

**Comment by rfennell** [comment]
Maybe, as this was designed for VSTS that is assumed and defaulted to defaultcollection, as you are on prem it needs to be added in (I did just write the sample URL from memory)

**Comment by gregpakes** [comment]
I have tried to solve this issue myself, but as a bit of a node rookie, I'm struggling to build the extension.  It seems I don't get a node_modules folder generated and therefore this does not get packaged up.  Therefore TFS fails to run due to missing dependencies....

**Comment by rfennell** [comment]
I assume this this will be fixed by the new 1.2.2 release with your PR in it

**Comment by gregpakes** [comment]
Sadly - it's not fixed by the PR.  I had to turn off BASIC auth on my server.  To be hones, that is almost certainly a best practice and I believe I only had it turned on because I was using early versions of the TFX tool which pre-dated "Personal Access Tokens".  There probably isn't much reason to have basic auth on a TFS server now.
